### PR TITLE
[WIP] Add pwsh ability for PowerShell task on windows, if installed

### PR DESCRIPTION
**Pull request by kilasuit**
Added checks for pwsh.exe & defaults to using pwsh on windows when it is available

### CODE REVIEW

**Comment by bergmeister** [comment]
Thanks for the first works. This PR needs to add at least a checkbox to optionally prefer pwsh over powershell because changing it to default/prefer pwsh would be a big breaking change.

File: Tasks/PowerShellV2/powershell.ps1 [hunk 1]
```diff
@@ -74,11 +74,22 @@ try {
     #
     # Note, use "-Command" instead of "-File". On PowerShell v4 and V3 when using "-File", terminating
     # errors do not cause a non-zero exit code.
+    #
+    # Adds in logic to try and use pwsh.exe first over powershell if installed on the agent.   
+    #
+    if ((Get-Command -name pwsh.exe -CommandType Application -ErrorAction SilentlyContinue).count -gt 0) 
+                    { $pwshPath = Get-Command -name pwsh.exe -CommandType Application | Select-Object -First 1 -ExpandProperty Path }
```
Comment (bergmeister) [inline Tasks/PowerShellV2/powershell.ps1]: The output of `Get-Command` should be stored in a temporary variable to not execute the command twice.

**Comment by kilasuit** [comment]
I will add in the `tick box` functionality as soon as I've read up more on the how to implement it. Any pointers on where to look would be useful :-)

**Comment by bergmeister** [comment]
@kilasuit I would add the checkbox in the `Advanced` section of it. You would add something like the following in the task json file [here](https://github.com/Microsoft/vsts-tasks/blob/master/Tasks/PowerShellV2/task.json#L107):
```
        {
            "name": "preferPwsh",
            "type": "boolean",
            "label": "Prefer PowerShell Core over Windows PowerShell on Windows",
            "required": false,
            "defaultValue": "false",
            "helpMarkDown": "If this is true, the task will look for the binary of PowerShell Core, which is named `pwsh.exe` is available on the PATH and if so use this as the main entry assembly. This applies to Windows only and if this is not true, it will default to Windows PowerShell (`powershell.exe`)",
            "groupName": "advanced"
        },
```
In the powershell script, you would receive this input then as follows:
```powershell
$input_preferPwsh = Get-VstsInput -Name 'preferPwsh' -AsBool
```

**Comment by bergmeister** [comment]
@kilasuit Sorry, this PR got superseded by  PR #7913 due to the lack of progress in this PR, which can be closed now.

**Comment by kilasuit** [comment]
Closing as per above apologies on delay been in middle of a house move glad this PR got superseeded in the meantime though! Thx @bergmeister

**Pull request closed without merging**